crates/rpc/rpc-eth-api/src/helpers/estimate.rs

```diff
@@ -81,28 +81,20 @@ pub trait EstimateCall: Call {
             apply_state_overrides(state_override, &mut db).map_err(Self::Error::from_eth_err)?;
         }
 
-        // Optimize for simple transfer transactions, potentially reducing the gas estimate.
-        if tx_env.input().is_empty() {
+        // Check if this is a basic transfer (no input data to account with no code)
+        let is_basic_transfer = if tx_env.input().is_empty() {
             if let TxKind::Call(to) = tx_env.kind() {
                 if let Ok(code) = db.db.account_code(&to) {
-                    let no_code_callee = code.map(|code| code.is_empty()).unwrap_or(true);
-                    if no_code_callee {
-                        // If the tx is a simple transfer (call to an account with no code) we can
-                        // shortcircuit. But simply returning
-                        // `MIN_TRANSACTION_GAS` is dangerous because there might be additional
-                        // field combos that bump the price up, so we try executing the function
-                        // with the minimum gas limit to make sure.
-                        let mut tx_env = tx_env.clone();
-                        tx_env.set_gas_limit(MIN_TRANSACTION_GAS);
-                        if let Ok(res) = self.transact(&mut db, evm_env.clone(), tx_env) {
-                            if res.result.is_success() {
-                                return Ok(U256::from(MIN_TRANSACTION_GAS))
-                            }
-                        }
-                    }
+                    code.map(|code| code.is_empty()).unwrap_or(true)
+                } else {
+                    false
                 }
+            } else {
+                false
             }
-        }
+        } else {
+            false
```

Comment:
we can make this a bit nicer and init the var with false and only set it to true when this is a transfer

Reply:
fixed sir! thank you :fire: